src/main/fc/config.c

```diff
@@ -267,7 +267,6 @@ static void setPidProfile(uint8_t pidProfileIndex)
     if (pidProfileIndex < MAX_PROFILE_COUNT) {
         systemConfigMutable()->pidProfileIndex = pidProfileIndex;
         currentPidProfile = pidProfilesMutable(pidProfileIndex);
-        pidInit(currentPidProfile); // re-initialise pid controller to re-initialise filters and config
```

Comment:
removing pidInit() from here will mean that when readEEPROM is called, after changing pid profile, e.g. using sticks, the pid and all the filters will not be correctly re-configured.

Check the call graph of setPidProfile:

![image](https://user-images.githubusercontent.com/57075/30783313-b480c3da-a138-11e7-8938-da0ea6368a8b.png)

Reply:
it feels like activateConfig() should call pidInit(), but that still won't fix the underlying problem on first boot.